src/if_statements.py

```diff
@@ -570,6 +603,216 @@ def try_make_if_condition(
+            # this node together with one outside the original interval. So, this node
+            # may fail the `start not in node.dominators` check below.
+            # Otherwise, treat these like empty BasicNodes.
+            nodes_to_mark_emitted.add(node)
+            if isinstance(node, BasicNode):
+                node_queue.append(node.successor)
+            continue
+
+        elif start not in node.dominators or end not in node.postdominators:
+            # The node must be within the [start, end] interval
+            pass
+
+        elif isinstance(node, BasicNode):
+            # This node has no statements, so it is just a goto
+            nodes_to_mark_emitted.add(node)
+            node_queue.append(node.successor)
+            continue
+
+        elif isinstance(node, ConditionalNode):
+            # If this is a "switch guard" `if` statement, continue iterating on both branches
+            control_expr = switch_guard_expr(node)
+            if (
+                control_expr is not None
+                and early_unwrap_ints(control_expr) == uw_var_expr
+            ):
+                node_queue.append(node.fallthrough_edge)
+                node_queue.append(node.conditional_edge)
+                nodes_to_mark_emitted.add(node)
+                continue
+
+            # Check if the branch_condition is a comparison between var_expr and a Literal
+            assert block_info.branch_condition is not None
+            cond = simplify_condition(block_info.branch_condition)
+            if (
+                isinstance(cond, BinaryOp)
+                and isinstance(cond.right, Literal)
+                and early_unwrap_ints(cond.left) == uw_var_expr
+            ):
+                # IDO typically uses `x == N` and `x < N` patterns in these if trees, but it
+                # will use `x != N` when it needs to jump backwards to an already-emitted block.
+                # GCC will freely use other types of comparisons
+                # Examples from PM: func_8026E558, pr_load_npc_extra_anims
+                if cond.op == "==":
+                    if cond.right.value in cases:
+                        return None
+                    cases[cond.right.value] = node.conditional_edge
+                    node_queue.append(node.fallthrough_edge)
+                elif cond.op == "!=" and (
+                    node.block.index > node.conditional_edge.block.index
+                    or context.options.compiler == context.options.CompilerEnum.GCC
+                ):
+                    if cond.right.value in cases:
+                        return None
+                    cases[cond.right.value] = node.fallthrough_edge
+                    node_queue.append(node.conditional_edge)
+                elif cond.op == "<" or (
+                    cond.op in (">=", "<=", ">")
+                    and context.options.compiler == context.options.CompilerEnum.GCC
```

Comment:
same question

Reply:
Thanks for checking - it looks like this one isn't needed. 
There *might* be a false-positive in MM's `Room_AllocateAndLoad` when I remove this check, but the structure is *very weird*, (removing the other `== GCC` check doesn't help with the outer `if` either)
```diff
     if ((temp_v0_5 != 0) && (temp_v0_5 != -2) && (temp_v0_5 != -7)) {
-        if ((temp_v0_5 == -8) || (temp_v0_5 == -5) || (temp_v0_5 == -4) || ((temp_v0_5 < 0) && (temp_v0_5 != -1) && (temp_v0_5 != -6))) {
-
-        } else {
+        switch (temp_v0_5) {                        /* irregular */
+        case -8:
+        case -5:
+        case -4:
+            break;
+        default:
             phi_v1_3 = temp_v0_5 - 1;
             if (temp_v0_5 < 0) {
                 phi_v1_3 = 2;
             }
+            break;
         }
         phi_a2 = (s32) gSaveContext.respawn[phi_v1_3].roomIndex;
     } else {
         phi_a2 = (s32) ((globalCtx + 0x18000)->unk_85C + ((globalCtx + 0x18000)->unk_B48 * 2))->unk_1;
     }
```